### PR TITLE
docs: Add instructions for local development

### PR DESCRIPTION
**Pull request by nguyen102**
Issue #, if available:

Description of changes:
* Added instructions and script to allow local development of the code.
* Added instructions and script to allow for packaging of local code and deployment to AWS

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/fhir-works-deploy.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -e
```
Comment (nguyen102) [inline scripts/fhir-works-deploy.sh:1]: I tried using `yarn` to add local packages, but `yarn` kept caching the local packages. Therefore, I was not always getting the latest packages when trying to deploy. Eventually I gave up using yarn local packages, and used yalc instead. [Reference](https://github.com/yarnpkg/yarn/issues/2165)

File: scripts/fhir-works-local-develop.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash -e
```
Comment (nguyen102) [inline scripts/fhir-works-local-develop.sh:1]: I tried using `yalc` links to do local development, but this is a very tedious process. Every time you make a change to a local package:
1. You have to do `yalc push` to push it to the local package manager. 
2. You have to remove all node modules in `development` package. 
3. You then relink the dependent packages in the `deployment` package. 
4. You then do yarn install.

Using `yarn` links is much faster.

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+`AWS_ACCESS_KEY_ID=<Access-Key> AWS_SECRET_ACCESS_KEY=<Secret-Key> OFFLINE_BINARY_BUCKET=<FHIRBinaryBucket> OFFLINE_ELASTICSEARCH_DOMAIN_ENDPOINT=<ElasticSearchDomainEndpoint> serverless offline start`
+
+If you don't know the `OFFLINE_BINARY_BUCKET` and `OFFLINE_ELASTICSEARCH_DOMAIN_ENDPOINT` value, you can run `serverless info --verbose` in the deployment package directory.
+
+## Deploy Local Packages to AWS
+
+If you have made changes to the `fhir-works-on-aws` packages, and you would like to deploy those changes to your AWS account, follow the instructions below.
+
+Before continueing with the rest of this section, be sure that you have followed the steps in the [**Development section**](#development) to set up your folder structure.
```
Comment (carvantes) [inline DEVELOPMENT.md]:
```suggestion
Before continuing with the rest of this section, be sure that you have followed the steps in the [**Development section**](#development) to set up your folder structure.
```

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# FHIR Works on AWS Development Instructions
+
+In this guide we will go over how to develop and run the code locally. We will also go over how to deploy your local code packages to your AWS account.
+
+## Prerequisites
+ 
+ - **Initial Install:** Please complete this [installation guide](./INSTALL.md) and get the initial `fhir-works-on-aws` code deployed to your AWS account.
+ - **Yalc**: We'll be using this to manage local packages, so please have have [this package](https://github.com/whitecolor/yalc) installed on your system.
```
Comment (carvantes) [inline DEVELOPMENT.md]:
```suggestion
 - **Yalc**: We'll be using this to manage local packages, so please have [this package](https://github.com/whitecolor/yalc) installed on your system. 
```

File: .gitignore [hunk 5]
```diff
@@ -23,3 +23,6 @@ coverage
 serverless-output.yml
 serverless-output.yaml
 serverless_config.json
+
+.yalc
+yalc.lock
```
Comment (carvantes) [inline .gitignore]: Add new line at end of file(It makes git happy). 

Usually eslint forces us to add new lines at the end of .ts files, but for other file types you need to manually add it.
You can set up your IDE to always add it https://intellij-support.jetbrains.com/hc/en-us/community/posts/207071445-On-save-new-line-added-to-end-of-file-How-stop-this-.

**Review by carvantes — DISMISSED** [review]

File: scripts/fhir-works-deploy.sh [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+yarn install
+yalc push
+
+cd ..
+
+printf "\nFor each package except for deployment and interface package, add interface package as a dependency, remove old pregenerated files, install all dependencies again, and push new packages to yalc\n"
+for i in "${packages[@]}"
+do
+    cd $i
+    rm -rf node_modules build/* dist/*
```
Comment (carvantes) [inline scripts/fhir-works-deploy.sh:29]: Is it necessary to delete node_modules and the re-run `yarn install`? This makes the whole process much slower. 

I know that yalc + serverless gets messy and starting on a clean slate makes things easier. But It'd be great to have this script run faster.

File: .gitignore (hunk 5 above)
Comment (nguyen102) [reply .gitignore]: added

File: DEVELOPMENT.md (hunk 4 above)
Comment (nguyen102) [reply DEVELOPMENT.md]: fixed

File: DEVELOPMENT.md (hunk 3 above)
Comment (nguyen102) [reply DEVELOPMENT.md]: fixed

File: scripts/fhir-works-deploy.sh (hunk 6 above)
Comment (nguyen102) [reply scripts/fhir-works-deploy.sh:29]: Without deleting `node_modules` and re running `yarn install`, the code would sometimes not pick up new changes to the dependent packages. Unfortunately, although this is slower, it's the most reliable way that I've found to make the deploy process work.

File: .gitignore [hunk 7]
```diff
@@ -25,4 +25,4 @@ serverless-output.yaml
 serverless_config.json
 
 .yalc
-yalc.lock
\ No newline at end of file
+yalc.lock
```
Comment (nguyen102) [inline .gitignore:28]:
```suggestion
yalc.lock

```

File: .gitignore (hunk 5 above)
Comment (nguyen102) [inline .gitignore:28]:
```suggestion
yalc.lock


```

**Review by carvantes — APPROVED** [review]

**Pull request merged**